nodebuilder/header/header.go

```diff
@@ -2,19 +2,22 @@ package header
 
 import (
 	"context"
+	"time"
 
 	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/peerstore"
+	"go.uber.org/fx"
 
 	"github.com/celestiaorg/celestia-node/header"
 	"github.com/celestiaorg/celestia-node/header/p2p"
 	"github.com/celestiaorg/celestia-node/header/store"
+	"github.com/celestiaorg/celestia-node/header/sync"
 	"github.com/celestiaorg/celestia-node/params"
 )
 
-// P2PExchange constructs new Exchange for headers.
-func P2PExchange(cfg Config) func(params.Bootstrappers, host.Host) (header.Exchange, error) {
+// newP2PExchange constructs new Exchange for headers.
+func newP2PExchange(cfg Config) func(params.Bootstrappers, host.Host) (header.Exchange, error) {
```

Comment:
I privatized these intentionally. Such proxy constructors should not be exported. I don't see a use case and we can export those if there is a request or so

cc @renaynay and @distractedm1nd